public/app/features/dashboard/components/SaveDashboard/SaveDashboardDrawer.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useMemo, useState } from 'react';
+import { css } from '@emotion/css';
+import { Checkbox, CustomScrollbar, Drawer, Tab, TabContent, TabsBar, useStyles2 } from '@grafana/ui';
+import { SaveDashboardData, SaveDashboardModalProps } from './types';
+import { GrafanaTheme2 } from '@grafana/data';
+import { jsonDiff } from '../VersionHistory/utils';
+import { selectors } from '@grafana/e2e-selectors';
+import { useAsync } from 'react-use';
+import { backendSrv } from 'app/core/services/backend_srv';
+import { useDashboardSave } from './useDashboardSave';
+import { SaveProvisionedDashboardForm } from './forms/SaveProvisionedDashboardForm';
+import { SaveDashboardErrorProxy } from './SaveDashboardErrorProxy';
+import { SaveDashboardAsForm } from './forms/SaveDashboardAsForm';
+import { SaveDashboardForm2 } from './forms/SaveDashboardForm2';
+import { SaveDashboardDiff } from './SaveDashboardDiff';
+
+export const SaveDashboardDrawer = ({ dashboard, onDismiss, onSaveSuccess, isCopy }: SaveDashboardModalProps) => {
+  const styles = useStyles2(getStyles);
+
+  const hasTimeChanged = useMemo(() => dashboard.hasTimeChanged(), [dashboard]);
+  const hasVariableChanged = useMemo(() => dashboard.hasVariableValuesChanged(), [dashboard]);
+
+  const [saveTimerange, setSaveTimerange] = useState(false);
+  const [saveVariables, setSaveVariables] = useState(false);
+
+  const status = useMemo(() => {
+    const isProvisioned = dashboard.meta.provisioned;
+    const isNew = dashboard.version === 0;
+    const isChanged = dashboard.version > 0;
+
+    return {
+      isProvisioned,
+      isNew,
+      isChanged,
+    };
+  }, [dashboard]);
```

Comment:
Not sure the purpose of this useMemo is, it's not doing anything expensive here. So would just move these to separate consts :)